service/src/main/java/bio/terra/externalcreds/services/ProviderClientCache.java

```diff
@@ -60,7 +60,7 @@ public ClientRegistration buildClientRegistration(
                     .toList()
                     .get(0);
             yield ClientRegistration.withRegistrationId(providerName)
-                .authorizationGrantType(AuthorizationGrantType.AUTHORIZATION_CODE)
```

Comment:
why did this change?

Reply:
In the [github docs for refreshing a user access](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/refreshing-user-access-tokens#refreshing-a-user-access-token-with-a-refresh-token) token it says the `grant_type` must be `refresh_token` to exchange the refresh for the access token

Reply:
And this doesn't impact the oauth login process where the Authorization Code is needed to complete the flow?

Reply:
ah I see, you're right looks like we need to have both at different times

Reply:
what do you think about building a new client registration where I'm getting the refresh token vs adding complexity to the cache?

Reply:
The cache is going to be very important to the performance of ECM. I think `TokenServiceProvider` might need its own cache.

Reply:
Looks like a `TokenClientCache`, analogous to `ProviderClientCache`, is what we want. Maybe renaming `ProviderClientCache` to `ProviderOAuthClientCache`, and creating a new `ProviderTokenClientCache`, is the way to go.

Reply:
Yeah that makes sense, this does seem like something we want cached. I can take a quick stab at it!